docs/src/api/Inference.md

```diff
@@ -0,0 +1,6 @@
+# API: `Turing.Inference`
+
+```@autodocs
+Modules = [Turing.Inference]
```

Comment:
ideally, this page is more structured. but I think we should merge this now  and fix it later.